frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/PipelineSuccessRatioDonut.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { ChartVoronoiContainer } from '@patternfly/react-charts';
+import { LoadingInline } from '@console/internal/components/utils';
+import { GraphEmpty } from '@console/internal/components/graphs/graph-empty';
+import { usePipelineSuccessRatioPoll } from '../hooks';
+import {
+  PipelineMetricsGraphProps,
+  getRangeVectorData,
+  formatDate,
+} from './pipeline-metrics-utils';
+import SuccessRatioDonut from './charts/successRatioDonut';
+import { TimeSeriesChart } from './charts/TimeSeriesChart';
+
+import './PipelineChart.scss';
+import { DEFAULT_CHART_HEIGHT } from '../const';
+
+const PipelineSuccessRatioDonut: React.FC<PipelineMetricsGraphProps> = ({
+  pipeline,
+  timespan,
+  interval,
+  width = 1000,
+}) => {
+  const {
+    metadata: { name, namespace },
+  } = pipeline;
+
+  const [runData, runDataError, runDataLoading] = usePipelineSuccessRatioPoll({
+    name,
+    namespace,
+    timespan,
+    delay: interval,
+  });
+
+  if (runDataLoading) {
+    return <LoadingInline />;
+  }
+  const pipelineRunTaskRunResultData = runData?.data?.result ?? [];
+  if (runDataError || pipelineRunTaskRunResultData.length === 0) {
+    return <GraphEmpty height={DEFAULT_CHART_HEIGHT} />;
+  }
+
+  const pipelineRuns = getRangeVectorData(runData, (r) => r.metric.status) ?? [];
+  const pipelineRunsCollection = pipelineRuns.reduce((acc, prun) => {
+    if (!prun) return acc;
+    const obj = prun[prun.length - 1];
+    acc.push(obj);
```

Comment:
Confused about this... each loop would push the same last item into the accumulator

Reply:
Basically it flattens the multi dimensional array by picking the last element in every iteration.

Reply:
Yup, not sure why I didn't see that last night